packages/utils/src/validators.ts

```diff
@@ -6,7 +6,46 @@ import {
   getNamespacesEventsForChainId,
   getAccountsChains,
 } from "./namespaces";
-import { hasOverlap, getDeletedDiff } from "./misc";
+import { getSdkError, getInternalError } from "./errors";
+import { hasOverlap } from "./misc";
+
+export type ErrorObject = { message: string; code: number } | null;
+
+// -- types validation ----------------------------------------------------- //
+
+export function isValidArray(arr: any, itemCondition?: (item: any) => boolean) {
+  if (Array.isArray(arr)) {
+    if (typeof itemCondition !== "undefined" && arr.length) {
+      const matches = arr.filter(itemCondition);
```

Comment:
Can simplify this by using `Array#every` to check every array item meets the predicate/condition, then no length comparison needed:

```suggestion
      return arr.every(itemCondition);
```